doc/src/modules/physics/mechanics/examples/multi_degree_freedom_holonomic_system.rst

```diff
@@ -3,88 +3,96 @@ Multi Degree of Freedom Holonomic System
-    ...                 joint_axis=compound_pend.z, parent_point=-l/3*compound_pend.y,
-    ...                 child_point=l*simple_pend.y)
-
-    >>> joints = (slider, rev1, rev2)
-
-Now we can apply loads (forces and torques) to the bodies, gravity acts on all bodies,
-a linear spring and damper act on block and wall, a rotational linear spring acts on C relative to B
-specified torque T acts on compound_pend and block, specified force F acts on block. ::
+In this case, the generalized coordinates :math:`q_1` represent lateral distance
+of block from wall, :math:`q_2` represents angle of the compound pendulum from
+vertical, :math:`q_3`  represents angle of the simple pendulum from the compound
+pendulum. The generalized speeds :math:`u_1` represents lateral speed of block,
+:math:`u_2` represents lateral speed of compound pendulum and :math:`u_3`
+represents angular speed of C relative to B.
+
+We also create some ``symbols`` to represent the length and mass of the
+pendulum, as well as gravity and others. ::
+
+   >>> from sympy import zeros, symbols
+   >>> from sympy.physics.mechanics import *
+   >>> q1, q2, q3, u1, u2, u3 = dynamicsymbols('q1, q2, q3, u1, u2, u3')
+   >>> l, k, c, g, kT = symbols('l, k, c, g, kT')
+   >>> ma, mb, mc, IBzz = symbols('ma, mb, mc, IBzz')
+
+Next, we create the bodies and an system for book-keeping.
+
+   >>> wall = RigidBody('N')
+   >>> system = System.from_newtonian(wall)
+   >>> block = Particle('A', mass=ma)
+   >>> IB = Inertia.from_inertia_scalars(Point('B_masscenter'),
+   ...                                   wall.frame, 0, 0, IBzz)
+   >>> compound_pend = RigidBody('B', IB.point, mass=mb, inertia=IB)
+   >>> simple_pend = Particle('C', mass=mc)
+
+Now we can connect the bodies using joints to establish the kinematics.
+Particles do not have frames from themselves, therefore we can either create a
+frame on beforehand or reuse a frame from the previous joint. After creating the
+joints we can add them to the system. ::
+
+
+   >>> slider = PrismaticJoint('J1', wall, block, coordinates=q1, speeds=u1)
+   >>> rev1 = PinJoint('J2', block, compound_pend, coordinates=q2, speeds=u2,
+   ...                 parent_interframe=slider.child_interframe,
+   ...                 joint_axis=wall.z, child_point=l*2/3*compound_pend.y)
+   >>> frame = ReferenceFrame('C_frame')
+   >>> rev2 = PinJoint('J3', compound_pend, simple_pend, coordinates=q3, speeds=u3,
+   ...                 joint_axis=compound_pend.z, parent_point=-l/3*compound_pend.y,
+   ...                 child_interframe=frame, child_point=l*frame.y)
+   >>> system.add_joints(slider, rev1, rev2)
+
+Now we can apply loads (forces and torques) to the bodies, gravity acts on all
+bodies, a linear spring and damper act on block and wall, a rotational linear
+spring acts on C relative to B specified torque T acts on compound_pend and
+block, specified force F acts on block. ::
```

Comment:
Struggling to make sense of this. Please reread.

Reply:
I didn't write this, but I'll update it to something readable.